### ISSUE TITLE
When pruning VGG16, the result is very bad while opening speed-up

### ISSUE THREAD

**Issue by iMurphL** [microsoft/nni#4402]
**Describe the issue**:
When pruning the model in the project, I found that speed up would make the results worse, so I intend to use vgg16 to reproduce the pruning results in cifar10 using the file "basic pruners_torch.py". The pruning method is L1 norm, sparsity is 0.5, and prune all convs, that is, I annotated line 260
![image](https://user-images.githubusercontent.com/38401357/146674078-90aed2e7-ab61-45f9-9d2e-51d2051ede84.png)
When speedup is false, finetune 1 epoch, and the model accuracy reaches 82%
When speedup is true, finetune 100 epoch, and the model accuracy is only 53%
![image](https://user-images.githubusercontent.com/38401357/146674199-1733825d-bc1d-4c9b-9cd3-2e2bfee5a6ed.png)
Apart from the influence of bias in BN when speedup is False, what else have I ignored? Could you tell me how the following results are reproduced? Thank you very much
![image](https://user-images.githubusercontent.com/38401357/146674242-7495ef34-a2d0-470d-80e6-1df3d95de28b.png)

**Environment**:
- NNI version: 2.3
- Training service (local|remote|pai|aml|etc): remote
- Client OS: Windows10
- Server OS (for remote mode only): Ubuntu18
- Python version: 3.6
- PyTorch/TensorFlow version: 1.6
- Is conda/virtualenv/venv used?: Yes
- Is running in Docker?: No

**Comment by J-shang** [comment]
Hello @iMurphL , I tried this setting in nni v2.5, and the results of finetuning 1 epoch are [no-speed-up 83.96] and [speed-up 84.8].

The reason for its performance so bad in v2.3 example is when we use speed-up, the modules in the model are replaced with the small ones, so the parameters handled by the old optimizer are deprecated. And v2.3 example has a bug, when we use speed-up, it uses the old optimizer to finetune the new compact model.

To fix it, you can initialize a new optimizer after speed-up for finetuning, and recommended using the latest nni version, it has a better speed-up experience.
